### PR TITLE
use annotation for exposed affinity for dataflow

### PR DESCRIPTION
**Pull request by xliuqq**
### Ⅰ. Describe what this PR does
user should define annotation for exposed dataflow affinity. see the usage in the doc.

### Ⅱ. Does this pull request fix one issue?

fixes #XXXX

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: docs/en/samples/dataflow_affinity.md [hunk 1]
```diff
@@ -97,15 +98,15 @@ When DataLoad B is running,  you will find the the affinity of its pod including
 
 The DataFlow consists of DataLoad A and DataLoad B. DataLoad A requires running on GPU nodes through customized label `node.kubernetes.io/instance-type`, while DataLoad B requires running on nodes with the same label value as DataLoad A (GPU nodes).
 
+- If DataLoad C (runAfter B) requires the affinity of this label, DataLoad B also needs to set the annotation `fluid.io/dataflow.affinity.labels: node.kubernetes.io/instance-type`
+
```
Comment (cheyang) [inline docs/en/samples/dataflow_affinity.md:102]: What about using the name `fluid.io/affinity-labels`? I think no need to add dataflow.

**Review by cheyang — CHANGES_REQUESTED** [review]

**Comment by TrafalgarZZZ** [comment]
@xliuqq @cheyang Is it possible to store affinity information in `annotations` instead of `labels`? From my understanding, annotations indicate which labels should be exposed, and dataflow-controller injects `key=value` pair to DataOperation job's labels. Can we also injecting `key=value` pair to DataOperation job's annotations?

File: pkg/common/label.go [hunk 2]
```diff
@@ -49,10 +49,13 @@ const (
 	// LabelNodePublishMothod is a pv label that indicates the method nodePuhlishVolume use
 	LabelNodePublishMothod = LabelAnnotationPrefix + "node-publish-method"
 
-	// AnnotationDataFlowAffinityInject is an annotation representing enabled the dataflow affinity injection
+	// AnnotationDataFlowAffinityInject is an annotation representing enabled the dataflow affinity injection, for internal use.
 	AnnotationDataFlowAffinityInject = LabelAnnotationPrefix + "dataflow-affinity.inject"
 	// LabelDataFlowAffinityPrefix is a prefix for customized dataflow affinity label name.
 	LabelDataFlowAffinityPrefix = "fluid.io."
```
Comment (TrafalgarZZZ) [inline pkg/common/label.go:55]: How about change the prefix to "dataflow-affinity.fluid.io."?

File: docs/en/samples/dataflow_affinity.md (hunk 1 above)
Comment (xliuqq) [reply docs/en/samples/dataflow_affinity.md:102]: ok, will fix.

**Comment by xliuqq** [comment]
> @xliuqq @cheyang Is it possible to store affinity information in `annotations` instead of `labels`? From my understanding, annotations indicate which labels should be exposed, and dataflow-controller injects `key=value` pair to DataOperation job's labels. Can we also injecting `key=value` pair to DataOperation job's annotations?

do you mean only store the affinity information in `annotations` instead of `labels` for DataOperation Job when it finished ?

File: pkg/common/label.go (hunk 2 above)
Comment (xliuqq) [reply pkg/common/label.go:55]: ok, will fix.

File: pkg/controllers/v1alpha1/fluidapp/dataflowaffinity/dataflowaffinity_controller.go [hunk 3]
```diff
@@ -155,47 +154,28 @@ func (f *DataOpJobReconciler) injectPodNodeLabelsToJob(job *batchv1.Job) error {
...
-	if pod.Spec.Affinity != nil && pod.Spec.Affinity.NodeAffinity != nil {
-		fillCustomizedNodeAffinity(pod.Spec.Affinity.NodeAffinity, injectLabels, node)
+	exposedLabels := pod.Annotations[common.AnnotationDataFlowAffinityLabelsName]
+	if len(exposedLabels) != 0 {
+		fillCustomizedNodeAffinity(injectLabels, node.Labels, strings.Split(exposedLabels, ","))
 	}
-
+	f.Log.Info("inject", "labels", exposedLabels)
 	// update job labels, reconciled job is selected by labels so the field will not be nil.
 	for k, v := range injectLabels {
```
[the remark this answers is not in the document]
Comment (xliuqq) [reply pkg/controllers/v1alpha1/fluidapp/dataflowaffinity/dataflowaffinity_controller.go:163]: ok.

**Comment by TrafalgarZZZ** [comment]
> > @xliuqq @cheyang Is it possible to store affinity information in `annotations` instead of `labels`? From my understanding, annotations indicate which labels should be exposed, and dataflow-controller injects `key=value` pair to DataOperation job's labels. Can we also injecting `key=value` pair to DataOperation job's annotations?
> 
> do you mean only store the affinity information in `annotations` instead of `labels` for DataOperation Job when it finished ? I agree with this.

Yes. All the affinity info DataFlow needs can be injected to annotations.

**Comment by cheyang** [comment]
> @xliuqq @cheyang Is it possible to store affinity information in `annotations` instead of `labels`? From my understanding, annotations indicate which labels should be exposed, and dataflow-controller injects `key=value` pair to DataOperation job's labels. Can we also injecting `key=value` pair to DataOperation job's annotations?

I also think annotation is better than labels. Thanks.

**Comment by cheyang** [comment]
@xliuqq Could you please fix the conflict of source file? Thanks.

File: sdk/python
Comment (TrafalgarZZZ) [inline sdk/python:1]: Please ignore updates on this file.

**Review by TrafalgarZZZ — CHANGES_REQUESTED** [review]

File: sdk/python
Comment (xliuqq) [reply sdk/python:1]: sorry for that. fixed.

**Review by cheyang — APPROVED** [review]
/lgtm
/approve

**Comment by TrafalgarZZZ** [comment]
fluid-e2e failed because the port allocator randomly picked the same port for two runtimes. It's a temporary failure
/retest

**Review by TrafalgarZZZ — APPROVED** [review]
/lgtm

**Pull request merged**